api/scm.go

```diff
@@ -164,14 +169,50 @@ func SyncRepo(c *gin.Context) {
 		// set repo to inactive - do not delete
 		r.SetActive(false)
 		// update repo in database
-		e := database.FromContext(c).UpdateRepo(r)
-		if e != nil {
+		err := database.FromContext(c).UpdateRepo(r)
+		if err != nil {
 			retErr := fmt.Errorf("unable to update repo for org %s: %w", org, err)
 
 			util.HandleError(c, http.StatusInternalServerError, retErr)
 
 			return
 		}
+		err = updateUserFavorites(c, r)
+		if err != nil {
+			util.HandleError(c, http.StatusInternalServerError, err)
+			return
+		}
 	}
 	c.JSON(http.StatusOK, fmt.Sprintf("repo %s synced", r.GetFullName()))
 }
+
+// updateUserFavorites is a helper function that takes a repo
+// that has been deleted from the SCM and ensures that it does
+// not appear in the user's favorites after its deletion.
+func updateUserFavorites(c *gin.Context, r *library.Repo) error {
+	// get user from the database
+	u, err := database.FromContext(c).GetUser(r.GetUserID())
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to retrieve user in database", baseErr)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+		return retErr
+	}
+
+	// if deleted repo exists in favorites, remove it
+	favorites := []string{}
+	for _, fav := range u.GetFavorites() {
+		if fav != r.GetFullName() {
+			favorites = append(favorites, fav)
+		}
+	}
+	u.SetFavorites(favorites)
+
+	// update the user favorites slice
+	err = database.FromContext(c).UpdateUser(u)
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to update repo %s: %v", baseErr, r.GetFullName(), err)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+		return retErr
+	}
+	return nil
```

Comment:
I think we're missing a space here:

```suggestion
	}
	
	return nil
```